### DISCUSSION TITLE
Does NVDA really announce changes in aria-describedby text or is it my implementation?

### DISCUSSION THREAD

**Discussion by AlmeroSteyn** [nvaccess/nvda#15250] [Q&A]
Thanks in advance for taking the time to read this. Will save me a lot of time building examples if I know the answer to this question.

I have noticed for a while now that if I have both aria-describedby to an error message as well as announce it with aria-live that I get double announcements in NVDA.

Today I did a test with a very basic example to trace this and came to the conclusion that if focus is set on the described element and the text in the combined containers referenced by aria-describedby changes, that the new text is announced directly.

Is it me doing something or is this expected NVDA behaviour?

And if so is there a way to control this to avoid duplicate announcements when combined with a live region for other screen readers?

**Comment by seanbudd** [comment]
It's difficult to understand this question without a sample. Would you be able to put a codepen together?

**Comment by AlmeroSteyn** [comment]
Of course! Sorry that it was not clear the first time around.

Here it is https://codepen.io/Kryptos_RSA/pen/eYQxjGq

I did it in VUE to approximate my environment and I wired it up so that each time you type a character in the input field the error message will show / hide with each keystroke. When this happens the aria-describedby changed to either only point to the id of the hint text or to both the hint text as well as the error text.

Each time NVDA announces the full combined text linked with aria-describedby.

Is this intended functionality?

**Comment by AlmeroSteyn** [comment]
Is there anything else required from my side to be able to answer this question? Thanks in advance!

**Comment by ABuffEr** [comment]
Hi @AlmeroSteyn,
I don't fully understand your example and situation, but consider that aria-describedby is to provide additional info about an element (e.g.: if you have a button with label "+", an aria-describedby may refer to a text like "to increase the counter").

**Comment by AlmeroSteyn** [comment]
Thanks for the reply @ABuffEr.

In short what I am experiencing is when keyboard focus is on the field with aria-describedby and, for example due to validation, the tekst linked to aria-describedby changes, NVDA auto announces as it of it a live region.
